enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+| --------------------------------- | ------------------------- | ---------------------------------------------------------------------------------- |
+| `.spec.pullspec`                  | Required                  | The fully-qualified digested image pullspec for the built OS image.                |
+| `.spec.baseOSImagePullspec`       | Required                  | The fully-qualified digested image pullspec that this OS image was built with.     |
+| `.spec.renderedMachineConfig`     | Required                  | The name of the rendered MachineConfig built into this image (if one is built in). |
+| `.spec.machineConfigPool`         | Required                  | The name of the MachineConfigPool this MachineOSImage was built for.               |
+
+If a cluster-admin were so inclined, they could potentially create their own
+MachineOSImage from an image built off-cluster. This could eventually replace
+the `osImageURL` mechanism provided today by MachineConfigs.
+
+#### Nodes (corev1.Node)
+
+The following annotations will be added to each node object to indicate
+the current and desired image states:
+
+| **Annotation Name:**                             | **Description:**                                                                                |
+| ------------------------------------------------ | ----------------------------------------------------------------------------------------------- |
+| `machineconfiguration.openshift.io/currentImage` | The fully-qualified digested image pullspec for the currently-booted OS image for a given node. |
+| `machineconfiguration.openshift.io/desiredImage` | The fully-qualified digested image pullspec for the desired OS image for a given node.          |
+
+**Discussion:** It was previously discussed that using the `osImageURL`
+field on the MachineConfigs will be a more consistent way of doing this in
+the future, so this practice will likely go away.
+
+#### MachineConfigPools
+
+The MachineConfigPool CRD will be extended thusly:
+
+| **Field Name / Path (JSONPath):** | **Optional or Required:** | **Intended Use:**                                                                                                                                                                                                |
+| --------------------------------- | ------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `.spec.machineOSImage`            | Required                  | The name of the MachineOSImage that the pool should be running. Changing this will roll out a new MachineOSImage to all of the nodes in the MachineConfigPool.                                                   |
```

Comment:
You cannot add a new required field to an already shipped API, that would be a breaking change